pkg/blobaccess/file/access.go

```diff
@@ -0,0 +1,338 @@
+package file
+
+import (
+	"io"
+	"sync"
+
+	"github.com/mandelsoft/goutils/errors"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"github.com/opencontainers/go-digest"
+
+	"github.com/open-component-model/ocm/pkg/blobaccess/bpi"
+	"github.com/open-component-model/ocm/pkg/iotools"
+	"github.com/open-component-model/ocm/pkg/utils"
+)
+
+type (
+	_nopCloser  = iotools.NopCloser
+	_blobAccess = bpi.BlobAccess
+)
+
+////////////////////////////////////////////////////////////////////////////////
+
+type fileDataAccess struct {
+	_nopCloser
+	fs   vfs.FileSystem
+	path string
+}
+
+var (
+	_ bpi.DataSource  = (*fileDataAccess)(nil)
+	_ bpi.Validatable = (*fileDataAccess)(nil)
+)
+
+func DataAccess(fs vfs.FileSystem, path string) bpi.DataAccess {
+	return &fileDataAccess{fs: fs, path: path}
+}
+
+func (a *fileDataAccess) Get() ([]byte, error) {
+	data, err := vfs.ReadFile(a.fs, a.path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "file %q", a.path)
+	}
+	return data, nil
+}
+
+func (a *fileDataAccess) Reader() (io.ReadCloser, error) {
+	file, err := a.fs.Open(a.path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "file %q", a.path)
+	}
+	return file, nil
+}
+
+func (a *fileDataAccess) Validate() error {
```

Comment:
Can we please have some short documentation? `Validate()` could mean a lot of different things in context of OCM... what's validated? Is the access correct? Does the files exist? Are the digests validated? ...?

Reply:
added